test/Microsoft.AspNet.Mvc.Localization.Test/HtmlLocalizerTest.cs

```diff
@@ -0,0 +1,106 @@
+using System;
+using System.Collections.Generic;
+using Xunit;
+
+namespace Microsoft.AspNet.Mvc.Localization.Test
+{
+    public class HtmlLocalizerTest
+    {
+        [Fact]
+        public void HtmlLocalizer_UseIndexer_ReturnsLocalizedString()
+        {
+            // Arrange
+            var localizedString = new LocalizedString("Hello", "Bonjour");
+            var stringLocalizer = new Mock<IStringLocalizer>();
+            stringLocalizer.Setup(s => s["Hello"]).Returns(localizedString);
+
+            var htmlLocalizer = new HtmlLocalizer(stringLocalizer.Object, new CommonTestEncoder());
+
+            // Act
+            var actualLocalizedString = htmlLocalizer["Hello"];
+
+            // Assert
+            Assert.Equal(localizedString, actualLocalizedString);
+        }
+
+        [Fact]
+        public void HtmlLocalizer_UseIndexerWithArguments_ReturnsLocalizedString()
+        {
+            // Arrange
+            var localizedString = new LocalizedString("Hello", "Bonjour test");
+
+            var stringLocalizer = new Mock<IStringLocalizer>();
+            stringLocalizer.Setup(s => s["Hello", "test"]).Returns(localizedString);
+
+            var htmlLocalizer = new HtmlLocalizer(stringLocalizer.Object, new CommonTestEncoder());
+
+            // Act
+            var actualLocalizedString = htmlLocalizer["Hello", "test"];
+
+            // Assert
+            Assert.Equal(localizedString, actualLocalizedString);
+        }
+
+        public static IEnumerable<object[]> HtmlData
+        {
+            get
+            {
+                yield return new object[] { "Bonjour {0} {{{{ }}", new object[] { "test" }, "Bonjour HtmlEncode[[test]] {{ }" };
+                yield return new object[] { "Bonjour {{0}}", new object[] { "{0}" }, "Bonjour {0}" };
+                yield return new object[] { "Bonjour {0:x}", new object[] { 10 }, "Bonjour HtmlEncode[[a]]" };
+                yield return new object[] { "Bonjour {0:x}}}", new object[] { 10 }, "Bonjour HtmlEncode[[x}]]" };
+                yield return new object[] { "Bonjour {{0:x}}", new object[] { 10 }, "Bonjour {0:x}" };
+                yield return new object[] { "{{ Bonjour {{{0:x}}}", new object[] { 10 }, "{ Bonjour {x}" };
+                yield return new object[] {
+                    "Bonjour {{{0:x}}} {1:yyyy}",
+                    new object[] { 10, new DateTime(2015, 10, 10) },
+                    "Bonjour {x} HtmlEncode[[2015]]"
+                };
+                yield return new object[] {
+                    "Bonjour {{{0:x}}} Bienvenue {{1:yyyy}}",
+                    new object[] { 10, new DateTime(2015, 10, 10) },
+                    "Bonjour {x} Bienvenue {1:yyyy}"
+                };
+                yield return new object[] {
+                    "Bonjour {0,6} Bienvenue {{1:yyyy}}",
+                    new object[] { 10, new DateTime(2015, 10, 10) },
+                    "Bonjour HtmlEncode[[    10]] Bienvenue {1:yyyy}"
+                };
```

Comment:
Extend this data to cover cases where "}}" appears _before_ the first "{" or "{{". Also test without "Bonjour " and with " Bonjour" after the last "}".